plugins/by-name/lsp-format/default.nix

```
@@ -91,28 +90,40 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.lsp-format" {
-      when = !config.plugins.lsp.enable;
+      when = !config.plugins.lspconfig.enable;
       message = ''
-        This plugin requires `plugins.lsp` to be enabled.
+        This plugin requires `plugins.lspconfig` to be enabled.
       '';
     };
 
-    plugins.lsp = {
-      onAttach =
-        lib.mkIf (cfg.lspServersToEnable == "all") # Lua
+    autoCmd = [
+      {
+        desc = "set up autoformatting on LSP attach";
+        event = "LspAttach";
+        callback = lib.mkIf (cfg.lspServersToEnable != "none") (
+          let
+            enableCondition =
+              if cfg.lspServersToEnable == "all" then
+                "true"
+              else if lib.isList cfg.lspServersToEnable then
+                # lua
+                ''
+                  vim.list_contains(${lib.nixvim.toLuaObject cfg.lspServersToEnable}, client.name)
+                ''
```

This avoids a trailing newline:

```suggestion
                "vim.list_contains(${lib.nixvim.toLuaObject cfg.lspServersToEnable}, client.name)"
```